backend/http_backend/http_backend.go

```diff
@@ -0,0 +1,114 @@
+package httpbackend
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/skroutz/downloader/job"
+)
+
+const DefaultClientTimeout = 30
+
+var (
+	// Based on http.DefaultTransport
+	//
+	// See https://golang.org/pkg/net/http/#RoundTripper
+	transport = &http.Transport{
+		Proxy: http.ProxyFromEnvironment,
+		DialContext: (&net.Dialer{
+			Timeout:   10 * time.Second, // was 30 * time.Second
+			KeepAlive: 30 * time.Second,
+			DualStack: true,
+		}).DialContext,
+		MaxIdleConns:          100,
+		IdleConnTimeout:       90 * time.Second,
+		TLSHandshakeTimeout:   10 * time.Second,
+		ExpectContinueTimeout: 1 * time.Second,
+	}
+
+	clientTimeout time.Duration
+)
+
+// HTTPBackend notifies about a job completion by executing an HTTP request.
+type HTTPBackend struct {
+	client  *http.Client
+	reports chan job.Callback
+}
+
+// ID returns "http"
+func (b *HTTPBackend) ID() string {
+	return "http"
+}
+
+// Start starts the backend based on configuration provided by cfg.
+func (b *HTTPBackend) Start(ctx context.Context, cfg map[string]interface{}) error {
+	cfgTimeout, ok := cfg["timeout"]
+	if !ok {
+		clientTimeout = time.Duration(DefaultClientTimeout) * time.Second
+	}
+	if ok {
+		clientTimeout = time.Duration(int(cfgTimeout.(float64))) * time.Second
+		if clientTimeout < (time.Duration(DefaultClientTimeout) * time.Second) {
+			clientTimeout = time.Duration(DefaultClientTimeout) * time.Second
```

Comment:
There was a comment instructed that client timeout must be greater than the dial timeout and the tls timeout. And since the client timeout was hardcoded 30 secs i figured this should be the default  and values smaller than that must be overriden.